classes/acrn-sblimage.bbclass

```diff
@@ -0,0 +1,88 @@
+inherit python3native
+DEPENDS += "slimboot-tools-native"
+
+MB_DEPENDENCY    ?= ""
+MB_MC_DEPENDENCY ?= ""
+MB_ACRN_BINARY   ?= "${TMPDIR}/deploy/images/${MACHINE}/acrn.32.out"
+MB_ACRN_CMDLINE  ?= ""
+MB_ACRN_MODULES  ?= "${TOPDIR}/conf/linux.txt;${TMPDIR}/deploy/images/${MACHINE}/${KERNEL_IMAGETYPE}"
+BASE_SBLIMAGE    ?= "sbl_os"
+SBLIMAGE_NAME    ?= "${BASE_SBLIMAGE}"
+PREGENERATED_SIGNING_KEY_SLIMBOOT_KEY_SHA256 ?= "${TOPDIR}/cert/TestSigningPrivateKey.pem"
+
+python do_acrn_sblimage() {
+    import re
+    import os
+
+    mbAcrnBinaryDeployDir = d.getVar('MB_ACRN_BINARY').lstrip().rstrip()
+    if not os.path.isfile(mbAcrnBinaryDeployDir):
+        bb.error("acrn %s not found!" % mbAcrnBinaryDeployDir)
+
+    mbAcrnCmdlineDeployDir = d.getVar('MB_ACRN_CMDLINE').lstrip().rstrip()
+    if mbAcrnCmdlineDeployDir == "":
+        hv_cmdline = d.getVar('WORKDIR') + "/hv_cmdline"
+        print("hv_cmdline: %s" % (hv_cmdline))
+        os.system("echo 'uart=mmio@0xfe042000' > %s" % (hv_cmdline))
+    else:
+        hv_cmdline = mbAcrnCmdlineDeployDir
+
+    mbAcrnBinaryDeployDirPairList = re.split(" +", d.getVar('MB_ACRN_MODULES').lstrip().rstrip())
+    print(*mbAcrnBinaryDeployDirPairList, sep = ", ")
+
+    genContainerPath = "%s/%s/slimboot/Tools/GenContainer.py" % (d.getVar('STAGING_DIR_NATIVE'), d.getVar('libexecdir'))
+
+    genContainerCmd = "%s" % d.getVar('PYTHON')
+    genContainerCmd = genContainerCmd + " %s create" % (genContainerPath)
+    genContainerCmd = genContainerCmd + " -cl"
+    genContainerCmd = genContainerCmd + " CMDL:%s" % (hv_cmdline)
+    genContainerCmd = genContainerCmd + " ACRN:%s" % (mbAcrnBinaryDeployDir)
+    counter = 0
+    for mbAcrnBinaryDeployDirPair in mbAcrnBinaryDeployDirPairList:
+        mbAcrnBinaryDeployDirPairMods = re.split(";", mbAcrnBinaryDeployDirPair.strip())
+
+        if not os.path.isfile(mbAcrnBinaryDeployDirPairMods[0]):
+            bb.error("module %s not found!" % mbAcrnBinaryDeployDirPairMods[0])
+
+        if not os.path.isfile(mbAcrnBinaryDeployDirPairMods[1]):
+            bb.error("module %s not found!" % mbAcrnBinaryDeployDirPairMods[1])
```

Comment:
Same comment, I don't know much about the internals ... when can this happen? Is there any helpful hint that we can provide here?

Reply:
Some module files such as Zephyr OS image are supposed to be provided by users. Because Yocto cannot build Zephyr. These errors can happen when the task could not find files specified by users. Again it will output file path which the task could not find as a hint of failure.

Reply:
Zephyr can be build using Yocto. We have meta-zephyr layer.
http://git.yoctoproject.org/cgit/cgit.cgi/meta-zephyr


Previously I created a PR to build zephyr as UOS. But need to rebase that PR.
https://github.com/intel/meta-acrn/pull/39